README.md

```diff
@@ -16,66 +16,69 @@ This repository is for Browser.html (the front-end). Active development of Graph
+If you're working on the Browser.html front-end, you'll need a [node][] to install all dependcies and run development toolchain.
 
 ``` sh
 git clone https://github.com/browserhtml/browserhtml.git
 cd browserhtml
 npm install --no-optional
 ```
 
-Then, start the front-end local server:
+Once all dependencies are installed, you can run a server component with following command:
 
 ``` sh
 npm run build-server
 ```
 
-Finally, in a new command line interface application (ex: Terminal), start Servo with the Browser.html flags turned on in either debug (`-d`) or release (`-r`) mode:
+Any changes to the source code will trigger a build, once complete server will serve a new version. That will allow you to reload a client in order to see your changes. Alternatively you could run a "live server" which also re-builds on changes, but in addition it also provides live code reload for the UI such that application state is preserves and no reload of the client is required. You can run "live server" with a following command:
 
-``` sh
-./mach run -r -- -b -w --pref dom.mozbrowser.enabled --pref dom.forcetouch.enabled --pref shell.builtin-key-shortcuts.enabled=false http://localhost:6060
+```sh
+npm run live-server
 ```
 
+### Client
 
-### Running in Gecko
+In order to run browser.html application itself, you will need a client component. This would be a Graphene runtime, built into a servo binary. You can either [download][servo download] pre-built nightly snapshots or [build it yourself][bulid servo] and run with `--browserhtml` flag. Assuming you have pre-built snapshot in default location on a Mac you can run browser.html application with a following command:
 
-Browser.html can also be run on top of a Gecko-based version of Graphene. We sometimes use this variant to test and debug features that haven't yet landed in Servo. Build instructions for Gecko-flavored Graphene can be found [on the wiki](https://github.com/browserhtml/browserhtml/wiki/Building-Graphene-%28Gecko-flavor%29).
 
+``` sh
+ /Applications/Servo.app/Contents/MacOS/servo -b -w --pref dom.mozbrowser.enabled --pref dom.forcetouch.enabled --pref shell.builtin-key-shortcuts.enabled=false http://localhost:6060
+```
 
-### Using WebIDE
-
-The easiest way to use developer tools with Browser.html is to select the "Remote Runtime" option in WebIDE while using the Gecko Graphene runtime.
+### Gecko Client
 
-By default you should be able to connect to the running browser at: localhost:6000.
+Browser.html can also be run on top of a Gecko-based version of Graphene. We used to use this variant to test and debug features that were not yet in Servo. You could either [download][dowload gecko] pre-build nightly snapshots or [build it yourself](https://github.com/browserhtml/browserhtml/wiki/Building-Graphene-%28Gecko-flavor%29). Assuming you have pre-build snapshot in default location on a Mac you can run browser.html appliaction with a following command:
```

Comment:
Very slightly fixed up:

Browser.html can also be run on top of a Gecko-based version of Graphene. We used to use this variant to test and debug features that were not yet in Servo. You can either [download][dowload gecko] pre-built nightly snapshots or [build it yourself](https://github.com/browserhtml/browserhtml/wiki/Building-Graphene-%28Gecko-flavor%29). Assuming you have a pre-built snapshot in default location on a Mac you can run the browser.html application with the following command: